src/Collapse/Collapse.component.tsx

```diff
@@ -108,32 +103,18 @@ export class Collapse extends React.Component<Props, any> {
 
   // Detect a new collapse state from props.isOpen change
   static getDerivedStateFromProps(props: Props, state: State) {
-    const isOpenProps =
-      props.isOpen !== undefined
-        ? !!props.isOpen && !props.isCollapsed
-        : !!props.isOpen || !props.isCollapsed;
-    const isOpen =
-      state.collapseState === EXPANDED || state.collapseState === EXPANDING;
-    const isCollapsed =
-      state.collapseState === COLLAPSED || state.collapseState === COLLAPSING;
-
-    const expand = !isOpenProps && isCollapsed;
-    const collapse = isOpenProps && isOpen;
+    const expand = !props.isOpen;
 
     if (expand) {
       return {
         hasReversed: state.collapseState === COLLAPSING,
```

Comment:
could be simplified?
return {
  hasReversed: props.isOpen ? state.collapseState === EXPANDING : state.collapseState === COLLAPSING,
  collapseState: props.isOpen? COLLAPSING : EXPANDING
}